### PR TITLE
chore: deprecate Trial/Experiment *_Reference classnames in Python SDK [MLG-858]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

Deprecate instead of remove old classnames `TrialReference` and `ExperimentReference` as to not cause a breaking change.

## Test Plan

- [ ] manually construct a `TrialReference` and `ExperimentReference` and verify at least one method still works on the class
```
from determined import Determined, experimental
client = Determined(master="localhost", user="username", password="******")
trial_ref = experimental.TrialReference(trial_id=1, session=client._session)
trial_ref.logs()


exp_ref = experimental.ExperimentReference(experiment_id=1, session=client._session)
exp_ref.list_trials()
```

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/common/experimental/experiment.py [hunk 1]
```diff
@@ -432,3 +432,25 @@ def _from_bindings(
...
+
+
+class ExperimentReference(Experiment):
+    """A legacy class representing an Experiment object.
+
+    This class was renamed to :class:`~determined.experimental.Experiment` and will be removed
+    in a future release.
+    """
+
+    def __init__(
```
Comment (wes-turner) [inline harness/determined/common/experimental/experiment.py:444]: nit: `__new__` is responsible for creating objects, not `__init__` (you can create a new `ExperimentReference` without calling `ExperimentReference.__init__`, but you can't create one without calling `__new__`). I think it's _slightly_ safer/more correct to override `__new__` for that reason.

**Review by wes-turner — APPROVED** [review]
Nice! I really like the method you came up with.

**Review by tara-det-ai — COMMENTED** [review]
viewed files without reviewing

File: harness/determined/common/experimental/experiment.py (hunk 1 above)
Comment (azhou-determined) [reply harness/determined/common/experimental/experiment.py:444]: discussed offline. new is too confusing.

**Pull request merged**